explore/src/main/scala/explore/ExplorePWA.scala

```diff
@@ -75,88 +82,83 @@ object ExplorePWA {
+        me.data.event match
+          case ExploreEvent.PWAReloadId      => IO.fromFuture(IO(updateSW(true).toFuture)) // .void
+          case ExploreEvent.ExploreUIReadyId => IO.fromFuture(IO(updateSW(()).toFuture))   // .void
+          case _                             => IO.unit
+      }
+      .compile
+      .drain
+  }
 
   // Setup the service worker
-  def setupSW(): Callback =
-    // Register the service worker and handle some callbacks
-    Callback {
-      val bc = new BroadcastChannel[ExploreEvent]("explore")
-
-      bc.onmessage = (
-        (x: ExploreEvent) =>
-          // This is coming from the client
-          x.event match {
-            case ExploreEvent.PWAReloadId      =>
-              IO(updateSW(true)).void
-            case ExploreEvent.ExploreUIReadyId =>
-              IO(updateSW).void
-            case a                             => IO.unit
-          }
-      ): (ExploreEvent => IO[Unit])
-
-      lazy val updateSW: js.Function1[js.UndefOr[Boolean], js.Promise[Unit]] =
-        registerSW(
-          RegisterSWOptions(
-            onNeedRefresh =
-              // If a new version is detected ask the usser
-              Callback.log("New version available, ask the user to update") *>
-                // Delay a bit to let the front setup the listener
-                requestUserConfirmation(bc),
-            onOfflineReady = Callback.log(s"Offline ready"),
-            onRegisterError = (x: js.Any) =>
-              Callback.log(s"Error on service worker registration $x") *> Callback(
-                org.scalajs.dom.window.console.log(x)
-              ),
-            onRegisteredSW = (u: String, r: ServiceWorkerRegistration) =>
-              import scala.concurrent.ExecutionContext.Implicits.global
-              // https://vite-plugin-pwa.netlify.app/guide/periodic-sw-updates.html
-              val isOffline =
-                !(!(r.installing && !js.isUndefined(org.scalajs.dom.window.navigator))) ||
-                  (dom.window.navigator.hasOwnProperty("connection") && dom.window.navigator.onLine)
-              Callback.log(
-                s"Service worker registered, setup self update task, offline: $isOffline"
-              ) *>
-                pingSW(u)
-                  .flatMap(go =>
-                    Callback(r.update())
-                      .delayMs(2000.0)
-                      .when(go)
-                      .attempt
-                  )
-                  .toCallback *>          // Inital check
-                scheduleUpdateCheck(u, r) // Periodic checks
-          )
-        )
-    }
+  def setupSW: IO[Unit] =
+    (for {
+      bc <- BroadcastChannel[IO, ExploreEvent]("explore")
+      _  <- Resource.eval(handleEvents(bc).start)
+    } yield ()).useForever
 
   @JSExport
   def runServiceWorker(): Unit =
-    setupSW().handleError(t => Callback(t.printStackTrace())).runNow()
+    setupSW.handleError(t => IO(t.printStackTrace())).unsafeRunAndForget()
```

Comment:
Same here, errors should be logged automatically.

```suggestion
    setupSW.unsafeRunAndForget()
```